src/AppInstallerCLICore/ContextOrchestrator.h

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include <AppInstallerLogging.h>
+#include "ExecutionReporter.h"
+#include "ExecutionArgs.h"
+#include "ExecutionContextData.h"
+#include "CompletionData.h"
+#include "COMContext.h"
+
+#include <string_view>
+
+namespace AppInstaller::CLI::Execution
+{
+    enum OrchestratorQueueItemState
+    {
+        Queued,
+        Running
+    };
+
+    struct OrchestratorQueueItem
+    {
+        OrchestratorQueueItem(OrchestratorQueueItemState state, std::shared_ptr<COMContext> context, HANDLE completedEvent) : m_completedEvent(completedEvent), m_state(state), m_context(context)  {}
+
+        OrchestratorQueueItemState GetState() { return m_state; }
+        void SetState(OrchestratorQueueItemState state) { m_state = state; }
+        std::shared_ptr<COMContext> GetContext() { return m_context; }
+        HANDLE GetCompletedEvent() { return m_completedEvent.get(); }
+    private:
+        OrchestratorQueueItemState m_state = OrchestratorQueueItemState::Queued;
+        std::shared_ptr<COMContext> m_context;
+        wil::unique_handle m_completedEvent;
+    };
+
+    struct ContextOrchestrator
+    {
+        static ContextOrchestrator& Instance();
+
+        std::shared_ptr<OrchestratorQueueItem> EnqueueAndRunCommand(std::shared_ptr<COMContext> context);

```

Comment:
True. I was originally writing this with more stuff defined for future Download\Remove support etc, but since none of that is actually supported yet it felt like it didn't make a ton of sense to try to define stuff that wasnt needed or used yet. I'll try again on naming.